K2Bridge/Visitors/Aggregations/DateRangeAggregationVisitor.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using System.Collections.Generic;
+    using System.Text;
+    using K2Bridge.Models.Request.Aggregations;
+
+    /// <content>
+    /// A visitor for the <see cref="DateRangeAggregation"/> element.
+    /// </content>
+    internal partial class ElasticSearchDSLVisitor : IVisitor
+    {
+        /// <inheritdoc/>
+        public void Visit(DateRangeAggregation dateRangeAggregation)
+        {
+            Ensure.IsNotNull(dateRangeAggregation, nameof(RangeAggregation));
+            EnsureClause.StringIsNotNullOrEmpty(dateRangeAggregation.Metric, nameof(RangeAggregation.Metric));
+            EnsureClause.StringIsNotNullOrEmpty(dateRangeAggregation.Field, nameof(RangeAggregation.Field));
+
+            string expandColumn = EncodeKustoField("_range_value");
+
+            // Extend expression:
+            // >> ['2']=pack_array("range1", "range2", "range3"), ['_range_value'] = pack_array(expr1, expr2, expr3)
+            // >> | mv-expand ['2'] to typeof(string), ['_range_value']
+            // >> | where ['_range_value'] == true
+            var extendExpression = new StringBuilder();
+
+            var rangeNames = new List<string>();
+            var rangeExpressions = new List<string>();
+
+            foreach (var range in dateRangeAggregation.Ranges)
+            {
+                range.Field = dateRangeAggregation.Field;
+                range.Accept(this);
+
+                if (string.IsNullOrEmpty(range.KustoQL))
+                {
+                    // This is then "open" range, -infinity to +infinity
+                    rangeExpressions.Add(KustoQLOperators.True);
+                }
+                else
+                {
+                    rangeExpressions.Add(range.KustoQL);
+                }
+
+                rangeNames.Add(range.BucketNameKustoQL);
+            }
+
+
+            extendExpression.Append($"{EncodeKustoField(dateRangeAggregation.Key)} = {KustoQLOperators.PackArray}({string.Join(',', rangeNames)}), ");
+            extendExpression.Append($"{expandColumn} = {KustoQLOperators.PackArray}({string.Join(',', rangeExpressions)})");
+
+            extendExpression.Append($"{KustoQLOperators.CommandSeparator} {KustoQLOperators.MvExpand} {EncodeKustoField(dateRangeAggregation.Key)} to typeof(string), {expandColumn}");
```

Comment:
Now DateRange, Range and Filter all have very similar code. Can we generalize it and avoid repetition?

Reply:
Yes, I agree. Especially code between Range and DateRange. But the associated Expression (RangeAggregationVisitor and DateRangeAggregationVisitor) are different. Not so evident to share this code. Do you have a suggestion ?

Reply:
Have BaseRangeAggregation that will have the `Field` and `Ranges` props, and move all of the logic to a function that will accept it? 

And if you want to go a step further make a base class that also includes Filter, with virtual members and methods to account for the small differences.

I really don't think the code should be duplicated 3 times, they are very very similar.

Reply:
OK, l work on this point.

Reply:
@AsafMah If possible, I prefer to postpone this refactoring until @tomconte implemented changes on filters. The same fix should be applied on Range and DateRange. We will probably have the same fix on all aggregations, but I prefer to be sure it will be the case before starting this code change.